extensions/mssql/src/mssql.d.ts

```diff
@@ -156,109 +156,22 @@ declare module 'mssql' {
 	export interface DacDeployOptionPropertyBoolean {
 		value: boolean;
 		description: string;
-		displayName: string;
+		propertyName: string;
```

Comment:
As I see this increasingly often, I'm getting more concerned with swap of `displayName to `propertyName`, when it's actually still the underlying `displayName`.  Is there a reason we don't want to use the actual property's name?

Reply:
We can use actual property names as key. 
Pros: easy to read [key:IgnoreColumnOrder; value:{true, "some description", "Ignore column order"}]
Cons: We need displayNames to be sorted as labels of options as shown in the ss. To get these, we need to go through the value to get the names as an array. when an option selection happened, we get the row index, using the row index we get the display name of the option from the array. to update the value boolean, we need to get the key, we can achieve it either by going through the values again or maintian an object of all the mappings [propertyNames:DisplayNames].

So I made this look like  [key:"Ignore column order"; value:{true, "some description", IgnoreColumnOrder(property will be used on STS side only)"}].
Pros: We can build the array using the keys(). using row index gets the name from array which is a key of optionsMapTable and can update the value directly.

@Benjin @Charles-Gagnon @kisantia @chlafreniere @llali @zijchen let me know if you see any better approaches that i can work on! thanks

Reply:
I don't think the key should be a display value, as called out before there could be issues if those values are localized.

When you create the table of options wouldn't you just take the array of options and then just loop over them and for each option create the checkbox and click handler that updates the model when clicked (using the current index of the option it's looking at)? Or is there another scenario you're thinking you need to do this mapping for?

Reply:
Since the options needs to be sorted alphabetically, onclick index doesn't match with the order of options. May be im misunderstanding your suggestion or explaining the same as you suggest. I'm thinking of an extra mapping {DisplayName:propertyName} to get the key for each index we get to update the map table.

Reply:
Regardless of that though - Benjin's point about changing displayName to propertyName seems confusing. Isn't the value contained here going to be the displayed name? As you mentioned the key is the property name itself - not this value.